### PR TITLE
Add peer reconnection mechanism

### PR DESCRIPTION
**Pull request by LukasGasior1**
Also added possibility to get peer list from PeerManager and status from Peer.
We try to reconnect when:
- we failed to connect to remote peer
- established connection was broken (most probably) i.e connection was closed but we did not receive (or send) Disconnect msg

### CODE REVIEW

**Comment by whysoserious** [comment]
Apart from the above it's a really good piece of code 👍

**Comment by LukasGasior1** [comment]
@whysoserious I'd really like to add tests for this but it doesn't seem trivial as PeerActor manages the underlying RLPxConnection child. I have not found satisfying solution for that so far.

**Comment by whysoserious** [comment]
Ok, we can write integration tests later.

File: src/main/scala/io/iohk/ethereum/network/PeerActor.scala [hunk 1]
```diff
@@ -131,54 +162,71 @@ class PeerActor(nodeInfo: NodeInfo) extends Actor with ActorLogging {
...
       }
+
     case DaoHeaderReceiveTimeout => disconnectFromPeer(rlpxConnection, Disconnect.Reasons.TimeoutOnReceivingAMessage)
+
+    case GetStatus => sender() ! StatusResponse(Handshaking)
+  }
+
+  private def disconnectFromPeer(rlpxConnection: RLPxConnection, reason: Int): Unit = {
+    rlpxConnection.sendMessage(Disconnect(Disconnect.Reasons.TimeoutOnReceivingAMessage))
+    system.scheduler.scheduleOnce(5.seconds, self, PoisonPill)
```
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/network/PeerActor.scala]: `5.seconds` to constant?

File: src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala [hunk 2]
```diff
@@ -10,25 +10,41 @@ class PeerManagerActor(nodeInfo: NodeInfo) extends Actor with ActorLogging {
...
+    case GetPeers =>
+      sender() ! peers
+
+    case Terminated(ref) =>
+      peers -= ref.path.name
   }
 
-  def createPeer(): ActorRef = {
+  def createPeer(addr: InetSocketAddress): Peer = {
     val id = UUID.randomUUID.toString
```
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala]: maybe use URI as `Id` ?

File: src/main/scala/io/iohk/ethereum/network/PeerActor.scala (hunk 1 above)
Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/network/PeerActor.scala]: @adamsmo maybe we can do a cleanup of this constants after https://github.com/input-output-hk/etc-client/pull/21 is merged ... I left some around too.

Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/network/PeerActor.scala]: @AlanVerbner sounds good
ok, so maybe we can add TODO comment to remember that this place needs to be changed

**Comment by LukasGasior1** [comment]
@whysoserious @AlanVerbner @adamsmo I've added unit tests for PeerActor reconnection logic. I've also added config object

**Comment by adamsmo** [comment]
👍 LGTM

**Comment by whysoserious** [comment]
LGTM 👍

**Pull request merged**